casper/src/main/scala/coop/rchain/casper/util/rholang/SystemDeploy.scala

```diff
@@ -23,6 +26,11 @@ abstract class SystemDeploy(val rand: Blake2b512Random) {
   protected def toEnvMap: ToEnvMap[Env]
   final def env: Map[String, Par] = normalizerEnv.toEnv(toEnvMap)
 
+  implicit protected val envsReturnChannel: Contains.Aux[Env, `sys:casper:return`, GUnforgeable]
+  protected def mkReturnChannel =
```

Comment:
It's interesting that it is critical that this is a definition.

Reply:
You might get NPE otherwise (though I didn't check). When a class is initialized on JVm it's superclasses are initialized first. Had this been `val`, `envsReturnChannel` would still be `null`. Boom.